crates/starknet-server/tests/test_simulate_transactions.rs

```diff
@@ -0,0 +1,403 @@
+    use starknet_rs_core::types::contract::SierraClass;
+    use starknet_rs_core::types::FieldElement;
+    use starknet_rs_core::utils::{
+        get_selector_from_name, get_udc_deployed_address, UdcUniqueness,
+    };
+    use starknet_rs_signers::Signer;
+
+    use crate::common::constants::{CAIRO_1_CONTRACT_PATH, CASM_COMPILED_CLASS_HASH, CHAIN_ID};
+    use crate::common::devnet::BackgroundDevnet;
+    use crate::common::utils::{
+        get_deployable_account_signer, get_predeployed_account_props, iter_to_hex_felt, load_json,
+        resolve_path, to_hex_felt, to_num_as_hex,
+    };
+
+    fn extract_overall_fee(simulation_result: &serde_json::Value) -> u128 {
+        let fee_hex = simulation_result["fee_estimation"]["overall_fee"].as_str().unwrap();
+        let fee_hex_stripped = fee_hex.strip_prefix("0x").unwrap();
+        u128::from_str_radix(fee_hex_stripped, 16).unwrap()
+    }
+
+    fn assert_fee_in_resp_greater(
+        resp_no_flags: &serde_json::Value,
+        resp_skip_validation: &serde_json::Value,
+    ) {
+        let no_flags_fee = extract_overall_fee(resp_no_flags);
+        let skip_validation_fee = extract_overall_fee(resp_skip_validation);
+        assert!(no_flags_fee.ge(&skip_validation_fee)); // TODO should be .gt, reported in https://github.com/lambdaclass/starknet_in_rust/issues/1051
+    }
+
+    fn assert_difference_if_validation(
+        resp_no_flags: &serde_json::Value,
+        resp_skip_validation: &serde_json::Value,
+        expected_contract_adddress: &str,
+    ) {
+        let no_flags_trace = &resp_no_flags["transaction_trace"];
+        assert_eq!(
+            no_flags_trace["validate_invocation"]["contract_address"].as_str().unwrap(),
+            expected_contract_adddress
+        );
+        assert!(no_flags_trace["fee_transfer_invocation"].as_object().is_none());
+
+        let skip_validation_trace = &resp_skip_validation["transaction_trace"];
+        assert!(skip_validation_trace["validate_invocation"].as_object().is_none());
+        assert!(skip_validation_trace["fee_transfer_invocation"].as_object().is_none());
+
+        assert_fee_in_resp_greater(resp_no_flags, resp_skip_validation);
+    }
+
+    #[tokio::test]
+    async fn simulate_declare_v1() {
+        let devnet = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+
+        // get account
+        let (signer, account_address) = get_predeployed_account_props();
+        let account = SingleOwnerAccount::new(
+            devnet.clone_provider(),
+            signer,
+            account_address,
+            CHAIN_ID,
+            ExecutionEncoding::Legacy,
+        );
+
+        let contract_json = dummy_cairo_0_contract_class();
+        let contract_artifact: Arc<LegacyContractClass> =
+            Arc::new(serde_json::from_value(contract_json.inner).unwrap());
+
+        let max_fee = FieldElement::ZERO; // TODO try 1e18 as u128 instead
```

Comment:
this TODO can be handled now

Reply:
As noted in the PR description, delegated to https://github.com/0xSpaceShard/starknet-devnet-rs/issues/192